py-scripts/lf_interop_video_streaming.py

```diff
@@ -740,41 +1166,44 @@ def generate_report(self, date, test_setup_info, realtime_dataset, report_path =
         for alias in eid_data["interfaces"]:
             for i in alias:
                 # Check interface index and alias
-                # alias[i]['mac'] alias[i]['ssid'] alias[i]['mode'] resource_hw_data['resource']['user']  
+                # alias[i]['mac'] alias[i]['ssid'] alias[i]['mode'] resource_hw_data['resource']['user']
                 if int(i.split(".")[1]) > 1 and alias[i]["alias"] == 'wlan0':
-                    
+
                     # Get resource data for specific interface
                     resource_hw_data = self.json_get("/resource/" + i.split(".")[0] + "/" + i.split(".")[1])
                     hw_version = resource_hw_data['resource']['hw version']
-                    
+
                     # Filter based on OS and resource ID
                     if not hw_version.startswith(('Win', 'Linux', 'Apple')) and int(resource_hw_data['resource']['eid'].split('.')[1]) in resource_ids:
                         device_type.append('Android')
-                        username.append(resource_hw_data['resource']['user'] )
+                        username.append(resource_hw_data['resource']['user'])
                         ssid.append(alias[i]['ssid'])
                         mac.append(alias[i]['mac'])
                         mode.append(alias[i]['mode'])
                         rssi.append(alias[i]['signal'])
                         channel.append(alias[i]['channel'])
-                        tx_rate.append(alias[i]['tx-rate']) 
-        total_urls = self.my_monitor('total-urls')
-        total_err = self.my_monitor('total-err')
-        total_buffer=self.my_monitor('total-buffers')
-        
+                        tx_rate.append(alias[i]['tx-rate'])
+        total_urls = self.data["total_urls"]
+        total_err = self.data["total_err"]
+        total_buffer = self.data["total_buffer"]
+        max_bytes_rd_list = []
+        avg_rx_rate_list = []
         # Iterate through the length of cx_order_list
-        for iter in range(len(cx_order_list)):
-            data_set_in_graph,wait_time_data,devices_on_running_state,device_names_on_running=[],[],[],[]
-            devices_data_to_create_wait_time_bar_graph=[]
-            max_video_rate,min_video_rate,avg_video_rate=[],[],[]
-            total_url_data,rssi_data=[],[]
-            trimmed_data_set_in_graph=[]
-             # Retrieve data for the previous iteration, if it's not the first iteration
-            if iter !=0:
-                before_data_iter=realtime_dataset[realtime_dataset['iteration']==iter]
+        for iter in range(len(iterations_before_test_stopped_by_user)):
+            data_set_in_graph, wait_time_data, devices_on_running_state, device_names_on_running = [], [], [], []
```

Comment:
Another scenario where a test object or something similar would be good. No change required.